src/go/rpk/pkg/kafka/client.go

```diff
@@ -171,6 +180,29 @@ func RetrySend(
 	return part, offset, err
 }
 
+func configureSASL(saramaConf *sarama.Config, rpConf *config.Config,
```

Comment:
do we want to add a test that given a `redpanda.yaml` file this gets configured correctly?

Reply:
I added for `LoadConfig` that verifies sarama configuration is correctly setup.